dp_check/dp_check.go

```diff
@@ -142,6 +227,35 @@ func main() {
 	if syscall.Getuid() != 0 {
 		infoLog.Println("Not running as root, some checks may fail.")
 	}
+
+	// Check if dp_check is running on a VM
+	runCheck("Running on GCE VM", func() error {
+		if !isRunningOnGCP() {
+			return fmt.Errorf(`dp_check is not running on a GCE VM`)
+		}
+		return nil
+	})
+
+	// Check connection to metadata server
+	runCheck("HTTP connectivity to metadata server", func() error {
+		client := &http.Client{}
+		metadataServerUrl := "http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/ipv6s"
+		req, err := http.NewRequest("GET", metadataServerUrl, nil)
+		if err != nil {
+			return err
+		}
+		req.Header.Add("Metadata-Flavor", "Google")
+		resp, err := client.Do(req)
+		infoLog.Printf("Check connection to metadata server by sending http GET request to %s", metadataServerUrl)
```

Comment:
note this check is less about HTTP connectivity to the metadata server than it is about checking that the current VM is DirectPath enabled (this API will return 200 with a response if so, and 404 if not). Can we add/change language to reflect that?